include/MySQL_Monitor.hpp

```diff
@@ -206,22 +216,73 @@ class MySQL_Monitor_State_Data {
 	int aws_aurora_add_lag_ms;
 	int aws_aurora_min_lag_ms;
 	int aws_aurora_lag_num_checks;
-  bool use_ssl;
-  MYSQL *mysql;
-  MYSQL_RES *result;
-  MYSQL *ret;
-  int interr;
-  char * mysql_error_msg;
-  MYSQL_ROW *row;
-  unsigned int repl_lag;
-  unsigned int hostgroup_id;
-	MySQL_Monitor_State_Data(char *h, int p, struct event_base *b, bool _use_ssl=0, int g=0);
+	bool use_ssl;
+	MYSQL *mysql;
+	MYSQL_RES *result;
+	int interr;
+	char *mysql_error_msg;
+	unsigned int repl_lag;
+	unsigned int hostgroup_id;
+	bool use_percona_heartbeat;
+	SQLite3DB* mondb;
+
+	MySQL_Monitor_State_Data(MySQL_Monitor_State_Data_Task_Type task_type, char* h, int p, bool _use_ssl = 0, int g = 0);
 	~MySQL_Monitor_State_Data();
-	SQLite3DB *mondb;
+	void init_async();
```

Comment:
Would be great to have some doc on the new functions added and the design changes to `MySQL_Monitor_State_Data`, the async nature of the tasks and the relation with `MySQL_Monitor_State_Data_Task_Result`.